# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsOpenURIInFrameParams.h
# Commit: fde2d124a1b1
# Full Hash: fde2d124a1b15877570390c7c69f5fcf6770640d
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-08 03:36:50
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 7.2: Create and use nsOpenWindowInfo types in nsWindowWatcher logic, r=kmag
#   
#   This patch builds on top of part 7.1 by creating this object within
#   nsWindowWatcher and ContentParent to carry the relevant information through
#   provider interfaces when creating new content windows. The nsOpenWindowInfo
# ==============================================================================

diff -r 7ccbbb127b7f -r fde2d124a1b1 dom/base/nsOpenURIInFrameParams.h
--- a/dom/base/nsOpenURIInFrameParams.h	Tue Apr 07 21:39:24 2020 +0000
+++ b/dom/base/nsOpenURIInFrameParams.h	Tue Apr 07 21:39:32 2020 +0000
@@ -26,14 +26,13 @@
   NS_DECL_CYCLE_COLLECTING_ISUPPORTS
   NS_DECL_NSIOPENURIINFRAMEPARAMS
 
-  explicit nsOpenURIInFrameParams(
-      const mozilla::OriginAttributes& aOriginAttributes,
-      mozilla::dom::Element* aOpener);
+  explicit nsOpenURIInFrameParams(nsIOpenWindowInfo* aOpenWindowInfo,
+                                  mozilla::dom::Element* aOpener);
 
  private:
   ~nsOpenURIInFrameParams();
 
-  mozilla::OriginAttributes mOpenerOriginAttributes;
+  nsCOMPtr<nsIOpenWindowInfo> mOpenWindowInfo;
   RefPtr<mozilla::dom::Element> mOpenerBrowser;
   nsCOMPtr<nsIReferrerInfo> mReferrerInfo;
   nsCOMPtr<nsIPrincipal> mTriggeringPrincipal;